cpp/src/io/orc/reader_impl_preprocess.cu

```diff
@@ -100,7 +100,8 @@ std::size_t gather_stream_info(std::size_t stripe_index,
 
   for (auto const& stream : stripefooter->streams) {
     if (!stream.column_id || *stream.column_id >= orc2gdf.size()) {
-      dst_offset += stream.length;
+      // Ignore reading this stream from source.
+      src_offset += stream.length;
```

Comment:
I think your change is correct, but we don't really have a way to test it.
Maybe we can add a warn log, since this implies invalid input.